src/epics/handleEmailLogin.js

```diff
@@ -61,43 +61,59 @@ const NEWSLETTER_SUBSCRIPTION_MUTATION = gql`
   }
 `
 
-export const handleLoginSuccess = (action$, store, { client }) =>
+export const handleLoginSuccess = action$ =>
   action$
     .ofType(actions.USER_LOGIN_SUCCESS)
     .mergeMap(action => {
       const { token, consent } = action
 
-      const hasSubscribed = localStorage.getItem(SUBSCRIPTION_FLAG)
-
-      if (hasSubscribed) {
-        localStorage.removeItem(SUBSCRIPTION_FLAG)
-      }
-
       return Observable.merge(
         Observable.of(showNotification('You are logged in!')),
-
         consent
           ? Observable.of(replace(`/consent?consent=${consent}&token=${token}`))
           : Observable.empty()
-      ).map(() =>
-        // NOTE(@vanguard): Delaying mutation because there is possible bug
-        // that appolo store has not updated and will consider user unathorized = mutation will fail
-        Observable.timer(2000).subscribe(() =>
-          hasSubscribed
-            ? client.mutate({
-              mutation: NEWSLETTER_SUBSCRIPTION_MUTATION,
-              variables: {
-                subscription: 'WEEKLY'
-              }
-            })
-            : undefined
-        )
       )
     })
     .catch(error => {
       return Observable.of({ type: actions.USER_LOGIN_FAILED, payload: error })
     })
 
+export const digestSubscriptionEpic = (action$, store, { client }) =>
+  action$
+    .ofType(actions.USER_LOGIN_SUCCESS)
+    .take(1)
+    .mergeMap(({ user: { privacyPolicyAccepted } }) =>
+      (privacyPolicyAccepted
+        ? Observable.of(true)
+        : action$.ofType(actions.USER_SETTING_GDPR)
+      )
+        .delayWhen(() => Observable.timer(2000))
```

Comment:
@DmitriVanGuard Why 2000?

Reply:
@Partysun If the request is sent immediately, the server might not be able to update internal data and will respond with `unauthorized` or `privacy policy is not accepted`. Happened several times while testing